classes/Village.php

```diff
@@ -129,4 +135,287 @@ public function getRelations(System $system, string $village_id): array {
+                $result = $system->db->fetch($result);
+                if ($result['elder_count'] == 3) {
+                    return "No seats available to claim!";
+                }
+                // check tier
+                if (!$player->reputation->rank >= 4) {
+                    return "You do not meet the reputation requirements!";
+                }
+                // check if has existing seat
+                $seat = self::getPlayerSeat($system, $player->user_id);
+                if (!empty($seat)) {
+                    return "You already have a seat in this village!";
+                }
+                // claim
+                $time = time();
+                $seat_title = "Elder";
+                $system->db->query("INSERT INTO `village_seats`
+                    (`user_id`, `village_id`, `seat_type`, `seat_title`, `seat_start`)
+                    VALUES ({$player->user_id}, {$player->village->village_id}, '{$seat_type}', '{$seat_title}', {$time})
+                ");
+                return "You have claimed the Elder seat!";
+                break;
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    public static function resign(System $system, User $player): string
+    {
+        $time = time();
+        $result = $system->db->query("UPDATE `village_seats` SET `seat_end` = {$time} WHERE `seat_end` IS NULL AND `user_id` = {$player->user_id}");
+        if ($system->db->last_affected_rows > 0) {
+            return "You have resigned!";
+        } else {
+            return "Something went wrong!";
+        }
+    }
+
+    /**
+     * @return array
+     */
+    public static function getVillageSeats(System $system, int $village_id): array {
+        $seats = [];
+        $seat_result = $system->db->query("SELECT `village_seats`.*, `users`.`user_name`, `users`.`avatar_link` FROM `village_seats`
+            INNER JOIN `users` on `village_seats`.`user_id` = `users`.`user_id`
+            WHERE `seat_end` IS NULL AND `village_id` = {$village_id} ORDER BY `seat_start` ASC");
+		$seat_result = $system->db->fetch_all($seat_result);
+        $elder_count = 0;
+        $has_kage = false;
+        foreach ($seat_result as $seat) {
+            switch ($seat['seat_type']) {
+                case 'kage':
+                    $has_kage = true;
+                    $seats['kage'] = [
+                        "seat_key" => 'kage',
+                        "seat_id" => $seat['seat_id'],
+                        "user_id" => $seat['user_id'],
+                        "village_id" => $seat['village_id'],
+                        "seat_type" => $seat['seat_type'],
+                        "seat_title" => $seat['seat_title'],
+                        "seat_start" => $seat['seat_start'],
+                        "user_name" => $seat['user_name'],
+                        "avatar_link" => $seat['avatar_link'],
```

Comment:
Hmm could you make a DTO for this? Looks like fields are same as the other seats below, would give us the benefits of static types in whatever code is calling this function

Reply:
Agreed, will refactor into a DTO.